packages/server/src/user/user.service.ts

```diff
@@ -86,12 +87,20 @@ export class UserService {
    * @returns `User` object, or throw `NotFoundError` when not exist
    */
   async findUserByUsername(projectId: string, username: string): Promise<User> {
-    return this.prisma.user.findFirstOrThrow({
-      where: {
-        projectId,
-        username
-      }
-    });
+    if (projectId == null || username == null) {
+      return null;
+    }
+
+    try {
+      return this.prisma.user.findFirst({
```

Comment:
Since the call isn't `findFirstOrThrow` this method won't throw an error so you shouldn't need the try catch.

Reply:
Good catch